t/proxyunits.t

```diff
@@ -69,9 +69,44 @@ for my $msrv (@mocksrvs) {
     push(@mbe, $be);
 }
 
-# Basic test with no backends. Write a command to the client, and check the
-# response.
+# $ps_send : request to proxy
+# $be_recv_ref : ref to a hashmap from be index to an array of received requests for validation.
+# $be_send_ref : ref to a hashmap from be index to an array of responses to proxy.
+# $ps_recv_ref : ref to response returned by proxy
+sub proxy_test {
```

Comment:
Having looked more closely at the tests I wonder about this a little more...

Passing in a hash for the expected backends means we lose order, and the function can't test order in which backends receive requests or the order in which it sends them back. Which is fine; this convenience function doesn't need to cover everything.

Except if all of the tests that use it are only ever talking to backend 0 you can cut most of the code out and shorten the text in the top level subtests?

Or did you have something else in mind here?

Reply:
Did three things:
1. Converted a test scenario to show how `proxy_test` can be used with multiple backends.
2. Added a comment about ordering `# backends in $be_recv and $be_send are vistied by looping through the @mbe.`
3. Loop @mbe to at least have a stable and predictable order of visiting.